x-pack/metricbeat/module/stan/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package subscriptions
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+
+	"github.com/elastic/beats/x-pack/metricbeat/module/stan"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/streaming/channelsz"
+	queryParams   = "subs=1"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+		PathConfigKey: "subscriptions.metrics_path",
+		QueryParams:   queryParams,
+	}.Build()
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("stan", "subscriptions", New,
+		mb.WithHostParser(hostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	http *helper.HTTP
+	Log  *logp.Logger
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	config := struct{}{}
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	http, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
+	}
+	return &MetricSet{
+		base,
+		http,
+		logp.NewLogger("stan"),
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(r mb.ReporterV2) (err error) {
+	_ = stan.AssetStan()
```

Comment:
What are we doing here?

Reply:
These calls look like dead code from a copy-pasta gone amok

Reply:
moving these!